File: app/src/main/java/com/amplifyframework/datastore/sample/LocalPresentation.java
package com.amplifyframework.datastore.sample;

import android.util.Pair;

import com.amplifyframework.datastore.generated.model.Post;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;
import io.reactivex.Single;

interface LocalPresentation {
    interface View {
        void displayLocalLogLine(LogLine logLine);
        void clearLocalLineItems();
    }

    interface Presenter {
        void createLocalItems();
        void updateLocalItems();
        void listLocalItems();
        void deleteLocalItems();
        void startSubscription();
        void clearLocalLog();
        void stopAllLocalActivities();
    }

    interface PostInteractor {
        Single<Post> createRandom();
        Single<List<Post>> updateAll(List<Post> posts);
        Single<List<Post>> list();
        Completable deleteAll(List<Post> posts);
        Observable<Pair<Post, Modification>> subscribe();
    }
}

File: app/src/main/java/com/amplifyframework/datastore/sample/LocalPresentation.kt
package com.amplifyframework.datastore.sample

import android.util.Pair

import com.amplifyframework.datastore.generated.model.Post

import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.Single

@Suppress("unused") // The sub-interface are, though. Lint doesn't catch that.
internal interface LocalPresentation {
    interface View {
        fun displayLocalLogLine(logLine: LogLine)
        fun clearLocalLineItems()
    }

    interface Presenter {
        fun createLocalItems()
        fun updateLocalItems()
        fun listLocalItems()
        fun deleteLocalItems()
        fun startSubscription()
        fun clearLocalLog()
        fun stopAllLocalActivities()
        fun signIn()
    }

    interface PostInteractor {
        fun createRandom(): Single<Post>
        fun updateAll(posts: List<Post>): Single<List<Post>>
        fun list(): Single<List<Post>>
        fun deleteAll(posts: List<Post>): Completable
        fun subscribe(): Observable<Pair<Post, Modification>>
        fun signIn(username: String, password: String): Completable
    }
}
